astroid/helpers.py

```diff
@@ -199,3 +199,44 @@ def class_instance_as_index(node):
     except exceptions.InferenceError:
         pass
     return None
+
+
+def object_len(node, context=None):
+    """Infer length of given node object
+
+    :param Union[nodes.ClassDef, nodes.Instance] node:
+    :param node: Node to infer length of
+
+    :raises AstroidTypeError: If an invalid node is given
+        for an invalid type is returned
+    :raises InferenceError: If the given node cannot be inferred
+        or if multiple nodes are inferred
+    :rtype int: Integer length of node
+    """
+    inferred_node = safe_infer(node, context=context)
+    if inferred_node is None or inferred_node is util.Uninferable:
+        raise exceptions.InferenceError(node=node)
+    if inferred_node.qname() in ('builtins.str', 'builtins.bytes'):
+        return len(inferred_node.value)
+    if 'elts' in inferred_node._astroid_fields:
+        return len(inferred_node.elts)
+    if 'items' in inferred_node._astroid_fields:
+        return len(inferred_node.items)
+    try:
+        node_type = object_type(inferred_node, context=context)
+        len_call = next(node_type.igetattr("__len__", context=context))
+    except exceptions.AttributeInferenceError:
+        raise exceptions.AstroidTypeError(
+            "object of type '{}' has no len()"
+            .format(len_call.pytype()))
+
+    try:
+        result_of_len = next(len_call.infer_call_result(node, context))
```

Comment:
The StopIteration is odd, we should never return it per se, but rather raise an astroid exception. Can you check what exactly is triggering a StopIteration?

Reply:
I'll double check, but I think the StopIteration is coming from classes that don't have a __len__ method

Reply:
next raises `StopIteration` when the given iterator is empty

Reply:
What I'm trying to say is that if someone does `next(node.infer_call_result())` that should either fail with InferenceError/return Uninferable or raise some other exception. It shouldn't fail with `StopIteration` explicitly, which usually shows a potential bug. If we put this in some other way, tools that use astroid, for instance pylint, should never care about catching `StopIteration` but only exceptions that astroid provide.

Reply:
Oh I see. infer_call_result shouldn't fail silently. I'm thinking it has to do with methods that are defined via the interpreter (written in C)